modules/numa-allocator/src/main/cpp/include/numa/numa_alloc.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _NUMA_ALLOC_H
+#define _NUMA_ALLOC_H
+
+#include <ostream>
+
+namespace numa {
+    class BitSet {
+    public:
+        BitSet();
+
+        BitSet(const BitSet &other);
+
+        BitSet &operator=(const BitSet &other);
+
+        ~BitSet();
+
+        class Reference {
```

Comment:
I don't quite get where this class is used. I see that it captures the position in a bitset and an object of the `Reference` is returned by the `[]` op. Is it used in the internals of the libnuma?

Reply:
Now it is not used. This is quite common technique  (see i.e. bitset implementation in [libstdc++|https://github.com/gcc-mirror/gcc/blob/85e91ad55a69282c1b0e34569836a026a1a954d1/libstdc%2B%2B-v3/include/std/bitset#L802]) to implement non-const `operator[]`. With it you can do somethin like this
```c++
BitSet a, b;
a[0] = true;
for (size_t i = 0; i < a.Size() && i < b.Size(); ++i) {
     a[i] = !b[i];
}
```
But this functionality is no used so far, it is for probably future uses